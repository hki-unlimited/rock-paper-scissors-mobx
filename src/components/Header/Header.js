import React from 'react';
import MainLogo from '../MainLogo/MainLogo';
import LocaleChanger from '../LocaleChanger/LocaleChanger';
import './Header.scss';

function Header() {
  return (
    <header className="dbs-app-header">
      <MainLogo />
      <LocaleChanger />
    </header>
  );
}

export default Header;
