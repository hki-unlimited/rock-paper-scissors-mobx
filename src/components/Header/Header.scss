header.dbs-app-header {
  display: flex;
  width: 100%;
  height: 6rem;
  align-items: center;
}
