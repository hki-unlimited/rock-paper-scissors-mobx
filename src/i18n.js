import i18n from 'i18next';
import translationFI from './assets/locales/fi.json';
import translationEN from './assets/locales/en.json';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

i18n
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources: {
      en: {
        translation: translationEN.en.translation
      },
      fi: {
        translation: translationFI.fi.translation
      }
    },
    lng: "en",
    fallbackLng: ["en", "fi"],
    whitelist: ["en", "fi"],

    interpolation: {
      escapeValue: false
    }
});

export default i18n