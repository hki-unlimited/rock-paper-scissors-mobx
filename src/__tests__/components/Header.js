import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '../../i18n';
import Header from '../../components/Header/Header';

afterEach(cleanup);

test("App header is rendered", async () => {
    const { container } = render(<Header />);
    expect(container.firstChild).toMatchSnapshot();
});

test("App header contains Schenker logo", async () => {
    const { container } = render(<Header />);
    const logo = await(() => container.querySelector(".dbs-main-logo"));
    expect(logo).toBeTruthy();
});