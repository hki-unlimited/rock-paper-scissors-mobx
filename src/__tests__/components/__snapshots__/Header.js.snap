// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App header is rendered 1`] = `
<header
  class="dbs-app-header"
>
  <a
    class="dbs-main-logo-link"
    href="https://www.dbschenker.com/fi-fi/"
    target="_blank"
  >
    <img
      alt="DB-Schenker"
      class="dbs-main-logo"
      src="db-schenker-logo.svg"
    />
  </a>
  <div
    class="dbs-logged-in-user-info"
  >
    Logged in as 
    <span
      class="dbs-logged-in-user-info-name"
    >
      Test User
    </span>
  </div>
  <div
    class="dbs-locale-changer"
  >
    <div
      class="dbs-locale-changer-locale-active"
    >
      EN
    </div>
    <div
      class="dbs-locale-changer-locale"
    >
      FI
    </div>
  </div>
</header>
`;
